### PR TITLE
Fix wrong entry in the document history when a new component is added.

### PR DESCRIPTION
**Pull request by jafarre-viewnext**

### CODE REVIEW

File: src/org/ods/orchestration/util/DocumentHistory.groovy [hunk 1]
```diff
@@ -227,6 +212,21 @@ class DocumentHistory {
         sortDocHistories(dhs).reverse()
     }
 
+    // Find the previous environment where the document was generated, if it exists.
+    // Otherwise, use the target environment.
+    @NonCPS
+    private getSourceEnvironment(String targetEnvironment, String documentName) {
+        def documentType = LeVADocumentUtil.getTypeFromName(documentName) as String
+        def environment = null
+        def envs = ['D', 'Q', 'P']
```
Comment (metmajer) [inline src/org/ods/orchestration/util/DocumentHistory.groovy:221]: @jafarre-viewnext should `envs` be defined in-line? Don't we have a global reference already?

**Review by metmajer — CHANGES_REQUESTED** [review]

**Review by metmajer — APPROVED** [review]

**Comment by metmajer** [comment]
@jafarre-viewnext this change actually needs to go into 4.x, not 3.x, no?

File: src/org/ods/orchestration/util/DocumentHistory.groovy (hunk 1 above)
Comment (jafarre-viewnext) [reply src/org/ods/orchestration/util/DocumentHistory.groovy]: I extracted the environments into an Enum

**Comment by jafarre-viewnext** [comment]
Some refactoring to make the code more maintainable has been performed

**Review by metmajer — APPROVED** [review]

**Comment by metmajer** [comment]
@jafarre-viewnext no need to add fixes into 3.x. Do we have a corresponding PR into master? Consider that what has a PR into master can be git merged directly into 4.x.

**Pull request merged**